api/scpca_portal/metadata_file.py

```diff
@@ -92,10 +92,29 @@ def write_metadata_dicts(list_of_dicts: List[Dict], output_file_path: str, **kwa
     Writes a list of dictionaries to a csv-like file.
     Optional modifiers to the csv.DictWriter can be passed to function as kwargs.
     """
-    kwargs["fieldnames"] = kwargs.get("fieldnames", utils.get_keys_from_dicts(list_of_dicts))
+
+    # TODO: Only pass the default utils.get_keys_from_dicts(list_of_dicts)
+    # to utils.get_sorted_field_names once Sample.Modalities.MULTIPLEXED is removed
+    kwargs["fieldnames"] = utils.get_sorted_field_names(
+        kwargs.get("fieldnames", utils.get_keys_from_dicts(list_of_dicts))
+    )
     kwargs["delimiter"] = kwargs.get("delimiter", common.TAB)
 
+    # Set default row sort if METADATA_ROW_SORT_ORDER[0] (i.e. scpca_project_id) in list_of_dicts
+    sorted_list_of_dicts = (
+        sorted(
+            list_of_dicts,
+            key=lambda k: (
+                k[common.METADATA_ROW_SORT_ORDER[0]],
+                k[common.METADATA_ROW_SORT_ORDER[1]],
+                k[common.METADATA_ROW_SORT_ORDER[2]],
+            ),
+        )
+        if any(common.METADATA_ROW_SORT_ORDER[0] in d for d in list_of_dicts)
```

Comment:
TODO: Remove when library model is implemented in file creation

We do want metadata files to all have the same base "combined" metadata ie those from project and sample